account-kit/infra/src/gas-manager.ts

```diff
@@ -26,44 +27,59 @@ export const ArbSepoliaPaymasterAddress =
 export const AlchemyPaymasterAddressV1 =
   "0xc03aac639bb21233e0139381970328db8bceeb67";
 
+export const AlchemyPaymasterAddressV07Mainnet =
+  "0x2cc0c7981D846b9F2a16276556f6e8cb52BfB633";
+export const AlchemyPaymasterAddressV07Testnet =
+  "0xEF725Aa22d43Ea69FB22bE2EBe6ECa205a6BCf5B";
+
 /**
  * Retrieves the Alchemy paymaster address for the given chain. Returns different addresses based on the chain ID.
  *
  * @example
  * ```ts
  * import { sepolia, getAlchemyPaymasterAddress } from "@account-kit/infra";
  *
- * const paymasterAddress = getAlchemyPaymasterAddress(sepolia);
+ * const paymasterAddress = getAlchemyPaymasterAddress(sepolia, "0.6");
  * ```
  *
  * @param {Chain} chain The chain for which the paymaster address is required
+ * @param {EntryPointVersion} version The version of the entry point
  * @returns {Address} The Alchemy paymaster address corresponding to the specified chain
- *
- * @deprecated This chain list in this function is no longer maintained since the ERC-7677 middleware is typically used to resolve the paymaster address
  */
-export const getAlchemyPaymasterAddress = (chain: Chain): Address => {
-  switch (chain.id) {
-    case polygonAmoy.id:
-    case optimismSepolia.id:
-    case baseSepolia.id:
-    case zora.id:
-    case zoraSepolia.id:
-    case fraxtal.id:
-    case fraxtalSepolia.id:
-      return AlchemyPaymasterAddressV3;
-    case mainnet.id:
-    case arbitrum.id:
-    case optimism.id:
-    case polygon.id:
-    case base.id:
-      return AlchemyPaymasterAddressV2;
-    case arbitrumSepolia.id:
-      return ArbSepoliaPaymasterAddress;
-    case sepolia.id:
-    case polygonMumbai.id:
-      return AlchemyPaymasterAddressV1;
+export const getAlchemyPaymasterAddress = (
+  chain: Chain,
+  version: EntryPointVersion,
+): Address => {
+  switch (version) {
+    case "0.6.0":
+      switch (chain.id) {
+        case polygonAmoy.id:
+        case optimismSepolia.id:
+        case baseSepolia.id:
+        case zora.id:
+        case zoraSepolia.id:
+        case fraxtal.id:
+          return AlchemyPaymasterAddressV3;
+        case mainnet.id:
+        case arbitrum.id:
+        case optimism.id:
+        case polygon.id:
+        case base.id:
+          return AlchemyPaymasterAddressV2;
+        case arbitrumSepolia.id:
+          return ArbSepoliaPaymasterAddress;
+        case sepolia.id:
+          return AlchemyPaymasterAddressV1;
+        default:
+          return AlchemyPaymasterAddressV4;
+      }
+    case "0.7.0":
+      if (chain.testnet) {
+        return AlchemyPaymasterAddressV07Testnet;
+      }
+      return AlchemyPaymasterAddressV07Mainnet;
```

Comment:
Because we know there will be an uncoming change to paymaster addresses on future networks, we should not support any default or catch-all cases in this PR.

For EP v0.6, we should avoid having a default case, and list out all current networks using `AlchemyPaymasterAddressV4`.

For EP v0.7, we should list out all testnet and mainnet chains using those addresses.

For both cases, the default should revert.